### PR TITLE
spike: DBTP-1058 Allow for building a Dotnet image

### PR DESCRIPTION
**Pull request by WillGibson**
Changes from spike to see how things would go building and deploying a Dotnet app on DBT Platform.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -17,7 +17,7 @@ INPUT_BUILDER="$BUILDPACKS_PATH/builder:$BUILDER_VERSION"
 BUILDER_RUN="$BUILDPACKS_PATH/run:$RUN_VERSION"
 LIFECYCLE="$ECR_PATH/buildpacksio/lifecycle:$LIFECYCLE_VERSION"
 BUILDPACK_JSON="buildpack.json"
-BUILDPACK_POST="fagiani/run@0.1.1"
+#BUILDPACK_POST="fagiani/run@0.1.1"
```
Comment (WillGibson) [inline build.sh:20]: This is not compatible with the newer version of the Paketo builder I used, we might want to try and resolve that.

File: image_builder/commands/deploy.py [hunk 2]
```diff
@@ -220,6 +220,16 @@ def clone_deployment_repository():
...
+    proc = subprocess.run(
+        "cd deploy && git checkout DBTP-1058-add-dotnet-service && cd ..",
+        stdout=subprocess.PIPE,
+        shell=True,
+    )
+
+    if proc.returncode != 0:
+        raise CannotCloneDeployRepositoryDeployError(
+            "Failed to checkout DBTP-1058-add-dotnet-service branch"
+        )
```
Comment (WillGibson) [inline image_builder/commands/deploy.py:232]: Temporary hack to use the spike branch.

File: image_builder/pack.py [hunk 3]
```diff
@@ -78,10 +80,11 @@ def get_command(self, publish=False):
         return command
 
     def get_buildpacks(self):
```
Comment (WillGibson) [inline image_builder/pack.py:81]: Various build packs that caused problems and were not necessary for the spike commented out.

File: image_builder/pack.py [hunk 4]
```diff
@@ -94,10 +97,14 @@ def get_buildpacks(self):
...
                 buildpacks.append("paketo-buildpacks/nodejs")
 
-        buildpacks.append("fagiani/run")
+        # buildpacks.append("fagiani/run")
         buildpacks.append("gcr.io/paketo-buildpacks/image-labels")
         buildpacks.append("gcr.io/paketo-buildpacks/environment-variables")
 
+        buildpacks.append("paketo-buildpacks/dotnet-core")
+        buildpacks.append("paketo-buildpacks/dotnet-publish")
+        buildpacks.append("paketo-buildpacks/dotnet-execute")
```
Comment (WillGibson) [inline image_builder/pack.py:105]: These were the three additional build packs needed to build the Dotnet image

**Comment by WillGibson** [comment]
We can resurrect this if we need it for reference, but it's gone stale now.

File: build.sh (hunk 1 above)
Comment (acodeninja) [reply build.sh:20]: I went and did it again, wrote yet another buildpack that replaces the run buildpack [acodeninja/run](https://github.com/acodeninja/buildpacks/tree/main/run) the cool thing with this is you can get rid of [load_run_environment.sh](https://github.com/uktrade/ci-image-builder/blob/main/image_builder/codebase/load_run_environment.sh) as the buildpack does all of that for you.

**Pull request closed without merging**